lib/pymedphys/_experimental/streamlit/apps/mosaiq_claude_chat/app.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (C) 2024 Simon Biggs
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import json
+import os
+import pathlib
+
+import httpx
+import trio
+from anthropic import AsyncAnthropic
+from anyio.from_thread import BlockingPortal
+from pymedphys._imports import streamlit as st
+
+import pymedphys
+from pymedphys._ai.sql_agent.messages import (
+    Messages,
+    message_content_as_plain_text,
+    receive_user_messages_and_call_assistant_loop,
+    write_message,
+)
+from pymedphys._mosaiq.server_from_bak import start_mssql_docker_image_with_bak_restore
+
+from ._trio import get_streamlit_trio_portal
+
+USER = "user"
+
+ANTHROPIC_API_LIMIT = 2
```

Comment:
```suggestion
# The default Anthropic plan limits API keys to only being able to call the API twice at the same time.
# Even this Mosaiq chat application tries to call the API up to 6 times concurrently (at least). This limiter
# constant is utilised to make it so that client side the API is only ever called 2 times at once. This limiter
# is not needed if your Anthropic plan does not have a max number of concurrent limit in place, and
# if your plan is not limited this application will run and work much faster without this limiter in place.
ANTHROPIC_API_LIMIT = 2
```

Reply:
Ooops, this one committed twice. I'll strip out the duplicate.